coderd/promoauth/oauth2_test.go

```diff
@@ -60,12 +60,23 @@ func TestInstrument(t *testing.T) {
 	// 0 Requests before we start
 	require.Nil(t, metricValue(t, reg, metricname, labels), "no metrics at start")
 
+	noClientCtx := ctx
+	// This should never be done, but promoauth should not break the default client
+	// even if this happens. So intentionally do this to verify nothing breaks.
+	ctx = context.WithValue(ctx, oauth2.HTTPClient, http.DefaultClient)
```

Comment:
This test failed before my change. It checks the prom metrics after making a call with the default http client. 

Will continue to see if I can cause this on local by adding some more chaos.

Reply:
I'd break this into its own test, but it's a good validation of the potential root cause 👍